pennylane/operation.py

```diff
@@ -1262,9 +1258,7 @@ def loc(w):
         if U.ndim == 1:
             W = np.zeros(dim)
             W[0] = U[0]
-            for k, w in enumerate(
-                self.wires.tolist()
-            ):  # TODO: refactor CV functions to not depend on consec wires
+            for k, w in enumerate(wire_indices):
```

Comment:
It seems that this function (and related ones) worked before by accessing a class attribute. How come we can't just make `wire_indices` an attribute instead of passing it around from function to function?

Reply:
I'm not sure what you mean exactly. `wire_indices` only gets passed from `heisenberg_obs` to `heisenberg_expand` (in the `CVObservable` class), or from `heisenberg_tr` to `heisenberg_expand` (in the `CVOperation` class). `heisenberg_tr` and `heisenberg_expand` get called by the qnode.

Of course, we could make those two functions set an attribute and then `heisenberg_expand` calls it. Is that what you mean? I always thought something like that is to be avoided, because it assumes that the attribute was always set before `heisenberg_expand` is called...But happy to do it that way!

Reply:
I was trying to understand why it was necessary to introduce a `wire_indices` arg to this function, when it wasn't needed before (from what I could determine [correct me if I'm wrong], this decision propagates to many other functions in `default.gaussian` then needing cosmetically updated signatures). I was wondering how we could avoid that. 

It seems before that this function accessed an attribute `self.wires.tolist()` rather than needing something to be passed. I was wondering if we could save all the redundant signature arguments that were introduced by using a similar trick here

Reply:
Unfortunately not. This is really the one ugly side effect of the refactor, which only surfaced when I dug deep into the `heisenberg_expand` function. 

But let me try to motivate while it is rather a case of uncovering a structure that was there all along than just making devices unnecessarily complicated:

The qnode calls `heisenberg_expand` to "transform" an operator it encounters (through `heisenberg_tr` and `heisenberg_obs`). But the transformation depends on where a wire is located on the device, because that determines what dimensions in the matrix to change. In other words, the transformation needs information on the device, and these functions are therefore not "local" to the operator class (which means you could just create an operator and then use them), but they require a device to make sense.

The previous "locality" was in some sense an artefact based on the assumption that the labels the operator knows of are locations of a wire on the device. But we are removing that assumption. So the function now needs to get information about the device, namely the register.